bsip-0087.md

```diff
@@ -0,0 +1,36 @@
+    BSIP: 0087
+    Title: Force Settlement Fee Ratio
+    Authors:
+      Jerry Liu bitcrab@qq.com
+    Status: Draft
+    Type: Protocol
+    Created: 2020-02-23
+    Discussion: https://github.com/bitshares/bsips/issues/260
+    Worker: TBD
+
+# Abstract
+This BSIP provide a solution to charge fee from force settlement.
+
+# Motivation
+Force settlement is an important part in smartcoin design, it provides the power to smartcoin holders to ask for collaterals with reference to feed price, and then guarantees the peg of smartcoin.
+
+The community continuously optimize the rules of smartcoin to balance the interests of relevant parties, taking bitCNY as an example, at the beginning the rules focus more on smart coin peg but care less on debt position owners’ interests, change are done to give debt position owners more protection, the force settlement offset has been changed from 0 to 2%, the max settlement  volume per maintenance period has been changed from no limit to 0.5% of smartcoin supply.  
+
+In the past several weeks, force settlement happened more intensively than before, the background is that the crypto market seemly switched from bear to bull, and BTS price went above 0.22CNY - which is the voted threshold per BSIP76, at some time the frequently happened force settlements made the lowest debt position collateral ratio even higher than 3, this phenomenon triggered warm discussion, almost all the users agree that the settler need to pay more cost while executing force settlement, and the cost will be paid to system as fee.
+
+# Rationale
+Force settlement fee can be seen as to pay part of the cost for smartcoin supply, stabilization and also the liquidation of collaterals, it is irrelevant to market fee sharing.
+To ensure the debt positions can always be closed with suffcient smartcoin, it is more feasible to cut off part of the traded collaterals as fee, instead of smartcoin.
+
+# Specification
+Add one new parameter Force Settlement Fee Ratio(FSFR) for each smartcoin, which is controlled by the smartcoin owner.
+`Force settlement order price = settlement price*(1+FSO)/(1-FSFR)`
+
+Here settlement price = feed price.
+
+FSO is Force Settlement Offset.
+
+When a force settlement is executed, the buyer sells smartcoin with quantity `X` and get collaterals in quantity `X*(1-FSFR)/[settlement price*(1+FSO)]`, the settled debt position owner sells collaterals in quantity `X/[settlement price*(1+FSO)]` and get smartcoin in quantity `X`, the delta in paid and received collaterals in quantity `X*FSFR/[settlement price*(1+FSO)]` will be paid to the owner of the smartcoin as force settlement fee.
```

Comment:
Can we add an example below where you use CNY ?

Reply:
> Can we add an example below where you use CNY ?

currently bitCNY is mainly used as one trading asset/derivative in DEX, it is very seldom used in scenarios other than this, I don't think we need to add CNY usage example here as this is just a BSIP.

Reply:
> 
> 
> > Can we add an example below where you use CNY ?
> 
> currently bitCNY is mainly used as one trading asset/derivative in DEX, it is very seldom used in scenarios other than this, I don't think we need to add CNY usage example here as this is just a BSIP.

This is just like a joke, if bitCNY is mainly used as one trading asset/derivative in DEX even currently, so why bitcny need it?

Why not use bitEUR add an example?

Reply:
> > > Can we add an example below where you use CNY ?
> > 
> > 
> > currently bitCNY is mainly used as one trading asset/derivative in DEX, it is very seldom used in scenarios other than this, I don't think we need to add CNY usage example here as this is just a BSIP.
> 
> This is just like a joke, if bitCNY is mainly used as one trading asset/derivative in DEX even currently, so why bitcny need it?
> 
> Why not use bitEUR to add an example? and if bitEUR is mainly used as one trading asset/derivative in DEX, so why we need this BSIP?

because Bitshares is now finding ways to get more income to pay necessary payments and buy back BTS. either force settlement or margin call is suitable for being charged.

and because bitEUR is in too low supply and too low trading volume to be a good example, if bitEUR is widely used somewhere outside BTS-DEX, please provide the details.

however, we need this BSIP even if bitEUR is widely used somewhere, this BSIP is to enable smartcoin owners to charge force settlement, whether to charge depend on smartcoin owner's choice.

Reply:
It's not about whether the chosen asset is a good cancidate. Call it bitABC if you want. It is about illustrating with actual number to give better understanding.

Reply:
> It's not about whether the chosen asset is a good cancidate. Call it bitABC if you want. It is about illustrating with actual number to give better understanding.

Thanks, updated the #Motivation part, please review.